### PR TITLE
CGMES export: tie lines as two separate equipment (ACLSs)

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

A `TieLine` (contains two paired `danglingLines`) is exported as a single equipment (ACLS).

**What is the new behavior (if this is a feature change)?**

The `TieLine` is exported as separated equipment (ACLS). The two paired `danglingLine` are exported as separated equipment, one ACLS by each `danglingLine`.

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractConductingEquipmentConversion.java [hunk 1]
```diff
@@ -372,6 +372,14 @@ protected static EquivalentInjectionConversion getEquivalentInjectionConversionF
         }
     }
 
+    // TODO graph is only valid if cgmes export preserves separated models
+    private EquivalentInjectionConversion getEquivalentInjectionConversionForDanglingLine(String graph, String boundaryNode) {
+        List<PropertyBag> eis = context.boundary().equivalentInjectionsAtNode(boundaryNode);
+
+        Optional<PropertyBag> optionalEi = eis.stream().filter(ei -> ei.get("graph").equals(graph)).findFirst();
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractConductingEquipmentConversion.java:379]: Please create a constant to replace the ```"graph"``` literal and comply with the sonar cloud check.

**Review by rcourtier — CHANGES_REQUESTED** [review]

**Comment by marqueslanauja** [comment]
I will update this PR after merging into the main the PR #2872 that simplifies creation of tie lines in assembled models (no subnetworks present).
Current implementation of this PR can be simplified afterPR #2872.
I prepared all the changes in a local branch that combines this PR and PR #2872

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportUtil.java [hunk 2]
```diff
@@ -273,14 +273,6 @@ public static String switchClassname(SwitchKind kind) {
     public static int getTerminalSequenceNumber(Terminal t, List<DanglingLine> boundaryDanglingLines) {
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/CgmesExportUtil.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java [hunk 3]
```diff
@@ -867,37 +855,14 @@ private static int getRatioTapChangerNeutralStep(RatioTapChanger rtc) {
...
-            }
-            String equivalentInjectionTerminalId = context.getNamingStrategy().getCgmesIdFromProperty(danglingLine, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "EquivalentInjectionTerminal");
-            if (!exported.contains(equivalentInjectionTerminalId)) { // check if the equivalent injection terminal has already been written (if several dangling lines linked to same X-node)
-                TerminalEq.write(equivalentInjectionTerminalId, equivalentInjectionId, connectivityNodeId, 1, cimNamespace, writer, context);
-                exported.add(equivalentInjectionTerminalId);
-            }
+            writeDanglingLineEquivalentInjection(danglingLine, cimNamespace, baseVoltageId, connectivityNodeId, exported, writer, context);
 
             // Cast the danglingLine to an AcLineSegment
             AcLineSegmentEq.write(context.getNamingStrategy().getCgmesId(danglingLine), danglingLine.getNameOrId() + "_DL",
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java]: Removed

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractConductingEquipmentConversion.java (hunk 1 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractConductingEquipmentConversion.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ACLineSegmentConversion.java [hunk 4]
```diff
@@ -82,22 +85,31 @@ public static void convertBoundaryLines(Context context, String boundaryNode, Bo
...
+        Optional.ofNullable(boundaryLine2.getBoundaryTopologicalNodeId()).ifPresent(tn -> mline.getDanglingLine2().setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TOPOLOGICAL_NODE_BOUNDARY, tn));
+        Optional.ofNullable(boundaryLine2.getBoundaryConnectivityNodeId()).ifPresent(cn -> mline.getDanglingLine2().setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.CONNECTIVITY_NODE_BOUNDARY, cn));
+
         context.convertedTerminal(boundaryLine1.getModelTerminalId(), mline.getDanglingLine1().getTerminal(), 1, boundaryLine1.getModelPowerFlow());
         context.convertedTerminal(boundaryLine2.getModelTerminalId(), mline.getDanglingLine2().getTerminal(), 2, boundaryLine2.getModelPowerFlow());
 
         context.terminalMapping().add(boundaryLine1.getBoundaryTerminalId(), mline.getDanglingLine1().getBoundary(), 2);
         context.terminalMapping().add(boundaryLine2.getBoundaryTerminalId(), mline.getDanglingLine2().getBoundary(), 1);
+
+        Optional.ofNullable(boundaryLine1.getEquivalentInjectionId()).ifPresent(eiId -> mline.getDanglingLine1().setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "EquivalentInjection", eiId));
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/ACLineSegmentConversion.java]: Done

**Review by zamarrenolm — APPROVED** [review]
Tested with CGM export in powsybl-entsoe

**Comment by rcourtier** [comment]
Just a question, when we export the TieLine in 2 ACLineSegment, I notice that the (boundary) ConnectivityNode isn't present in the exported CGMES EQ file. Is that intended? I understand it originally comes from the boundary EQ file defined by ENTSO-E, but I was wondering if we wouldn't want to include it in the export of the merged network?

**Review by rcourtier — APPROVED** [review]

**Pull request merged**